graph/quadstore.go

```diff
@@ -125,119 +125,52 @@ type QuadStore interface {
 
 type Options map[string]interface{}
 
+var (
+	typeInt = reflect.TypeOf(0)
+)
+
 func (d Options) IntKey(key string) (int, bool, error) {
 	if val, ok := d[key]; ok {
-		switch vv := val.(type) {
-		case float64:
-			return int(vv), true, nil
-		default:
-			return 0, false, fmt.Errorf("Invalid %s parameter type from config: %T", key, val)
+		if reflect.TypeOf(val).ConvertibleTo(typeInt) {
+			return val.(int), true, nil
 		}
+
+		return 0, false, fmt.Errorf("Invalid %s parameter type from config: %T", key, val)
 	}
+
 	return 0, false, nil
 }
 
 func (d Options) StringKey(key string) (string, bool, error) {
 	if val, ok := d[key]; ok {
-		switch vv := val.(type) {
-		case string:
-			return vv, true, nil
-		default:
-			return "", false, fmt.Errorf("Invalid %s parameter type from config: %T", key, val)
+		if v, ok := val.(string); ok {
+			return v, true, nil
 		}
+
+		return "", false, fmt.Errorf("Invalid %s parameter type from config: %T", key, val)
 	}
+
 	return "", false, nil
 }
 
 func (d Options) BoolKey(key string) (bool, bool, error) {
 	if val, ok := d[key]; ok {
-		switch vv := val.(type) {
-		case bool:
-			return vv, true, nil
-		default:
-			return false, false, fmt.Errorf("Invalid %s parameter type from config: %T", key, val)
+		if v, ok := val.(bool); ok {
+			return v, true, nil
 		}
+
+		return false, false, fmt.Errorf("Invalid %s parameter type from config: %T", key, val)
 	}
+
 	return false, false, nil
 }
 
-var ErrCannotBulkLoad = errors.New("quadstore: cannot bulk load")
-var ErrDatabaseExists = errors.New("quadstore: cannot init; database already exists")
+var ErrCannotBulkLoad = fmt.Errorf("quadstore: cannot bulk load")
```

Comment:
Leave it as `errors.New`. It's not a good practice to use `fmt.Errorf` if you don't need to.